crates/turbopack-ecmascript/src/references/esm/export.rs

```diff
@@ -147,6 +140,18 @@ impl CodeGenerateable for EsmExports {
                         );
                     }
                 }
+
+                if let EcmascriptExports::CommonJs = &*asset.get_exports().await? {
```

Comment:
Decide based on the boolean flag returned by expand_star_exports if `__turbopack__cjs__` is needed.

This is needed when reexporting CJS with multiple layers:

``` js
// commonjs.js
exports.hello = "World";

// c.js
export * from "./commonjs.js";
// This would be handled by existing logic

// b.js
export * from "./c";
// This would not be handled, but still need __turbopack__cjs__
// as there are properties dynamically added by __turbopack__cjs__ in c.js

// a.js
import * as B from "./b";
console.log(B);
```